oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -116,11 +116,11 @@ public List<OSProcess> getChildProcesses(int parentPid, Predicate<OSProcess> fil
         // Get this pid and its children
         List<OSProcess> childProcs = queryChildProcesses(parentPid);
         // Extract the parent from the list
-        OSProcess parent = childProcs.stream().filter(p -> p.getProcessID() == parentPid).findAny().orElse(null);
+        OSProcess parent = childProcs.stream().parallel().filter(p -> p.getProcessID() == parentPid).findAny().orElse(null);
         // Get the parent's start time
         long parentStartTime = parent == null ? 0 : parent.getStartTime();
         // Get children after parent
-        return queryChildProcesses(parentPid).stream().filter(filter == null ? ALL_PROCESSES : filter)
+        return queryChildProcesses(parentPid).stream().parallel().filter(filter == null ? ALL_PROCESSES : filter)
```

Comment:
Probably very little value in the parallel filters, and concern about nondeterministic results with a limit at the end.